src/main/java/hudson/plugins/dimensionsscm/DimensionsAPICallback14.java

```diff
@@ -125,18 +128,22 @@ public void saveChangesToXmlFile(DimensionsAPI dimensionsAPI, DimensionsConnecti
         }
     }
 
-    private boolean notMoveOrDeleteOpType(STEP_TYPE stepType){
+    private boolean notMoveOrDeleteOpType(STEP_TYPE stepType) {
         return !stepType.equals(STEP_TYPE.REMOVE) && !stepType.equals(STEP_TYPE.MOVE);
     }
 
-    private void createChangeListFromChangeSteps(DimensionsAPI dimensionsAPI, List<DimensionsChangeStep> dimensionsChangeSteps, TimeZone tz, final String url, Map<String, DimensionsChangeLogEntry> entries, List<ItemRevision> items) {
+    private void createChangeListFromChangeSteps(DimensionsAPI dimensionsAPI, FilePath projectDir, List<DimensionsChangeStep> dimensionsChangeSteps, TimeZone tz, final String url, Map<String, DimensionsChangeLogEntry> entries, List<ItemRevision> items) {
 
         Map<Long, ItemRevision> itemRevisionToUidMap = createItemRevisionMap(items);
 
         for (DimensionsChangeStep changeStep : dimensionsChangeSteps) {
 
+            String projectPath = PathUtils.normalizeSlashes(changeStep.getProjectPath());
+            String remotePath = PathUtils.normalizeSlashes(projectDir.getRemote());
+            if (!projectPath.contains(remotePath))
+                continue;
```

Comment:
What does this do? Don't understand why the changelist would be affected by agent

Reply:
Agent don't affect changeList.
This is fix for DEF337415 
projectDir.getRemote() just get folder which was set in Folders field by user.
![image](https://user-images.githubusercontent.com/32028221/72348568-5a60f080-36e3-11ea-8fe0-782e28ca8ec3.png)

This check tell: if path from changeset isn't fit to paths which user set to Folders field than just skip it.
Maybe it just bad name for variable.

Reply:
Right 👍